Backpack-SwiftUI/SponsoredBanner/Classes/BPKSponsoredBanner.swift

```diff
@@ -0,0 +1,201 @@
+//
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright © 2023 Skyscanner Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.a=pache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKSponsoredBanner: View {
+    let logo: Image?
+    let title: String?
+    let subheadline: String?
+    let callToAction: CallToAction?
+    let bodyText: String?
+    let variant: Variant
+    let backgroundColor: Color
+    let customAccessibilityLabel: String?
+    
+    @State var showTerms: Bool
+    
+    public init(
+        logo: Image? = nil,
+        title: String? = nil,
+        subheadline: String? = nil,
+        callToAction: CallToAction? = nil,
+        bodyText: String? = nil,
+        variant: Variant,
+        backgroundColor: Color,
+        showTerms: Bool = false,
+        customAccessibilityLabel: String? = nil
+    ) {
+        self.logo = logo
+        self.title = title
+        self.subheadline = subheadline
+        self.callToAction = callToAction
+        self.bodyText = bodyText
+        self.variant = variant
+        self.backgroundColor = backgroundColor
+        self.showTerms = showTerms
+        self.customAccessibilityLabel = customAccessibilityLabel
+    }
+    
+    public var body: some View {
+        VStack(spacing: 0) {
+            topView
+                .background(backgroundColor)
+                .zIndex(1)
+            bodyTextView
+            
+        }
+        .background(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+                .foregroundColor(.sponsoredBannerBackgroundColor)
+        )
+        .clipShape(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+        )
+        .if(customAccessibilityLabel != nil) { view in
+            view
+                .accessibilityElement(children: .ignore)
+                .accessibilityLabel(customAccessibilityLabel ?? "")
+        }
+    }
+    
+    private var bodyTextView: some View {
+        Group {
+            if hasBody && showTerms == true {
+                BPKText(bodyText ?? "", style: .caption)
```

Comment:
I think you probably want to also if-let the body text as well, to not accidentally show an empty BPKText